### ISSUE TITLE
Guards scripting.

### ISSUE THREAD

**Issue by Cabro** [elysium-project/server#2424]
It always makes me wonder why all private servers have the check for pvp activities at the source player, not at the target player or even better, at both. This makes it possible for safespotting where it wasnt impossible on retail.

**Comment by Mostprovidedlinks** [comment]
all safespotting should be fixed and prevented. I think safespotting is stupid

**Comment by Phoosy** [comment]
Before opening any bug reports on the issue tracker, please read the guidelines [located here](https://github.com/elysium-project/server/blob/development/docs/).

**Comment by Cabro** [comment]
The check for pvp activities should be also done at the target player.
This good?

**Comment by thepparker** [comment]
So... attacking someone should get them killed by the guards? 🤔

**Comment by Cabro** [comment]
No. If you would make the pvp check also at the target player (as in the one that is getting ganked from someone safespotting), the system would detect that the attacks come from the ganker, and is making the guards attack the ganker, with shoot/net/kick whatever. Because it can also be done that the attacker is simply out of range of the guards, and they will never detect pvp activities, since nobody that is in their range is doing harmful attacks on players. You are only checking for the players that are causing the "gank" to be in range of the guards, but this is wrong. You should also check if the victim is range also, because right now you can get nullufied right in front of a guard and it will do nothing, becuase there is no pvp action for them.

**Comment by Phoosy** [comment]
http://www.wowhead.com/forums&topic=32255#p392776

It would appear (at least as of patch Patch 3.0.1) that guards behave as they do currently on Elysium based on this comment. Obviously not up to snuff in terms of actual evidence, but the current behavior _appears_ to be normal operating behavior for guards.

**Comment by Cabro** [comment]
What you found there is true for the Isle of Quel'Dalas guads only. Read all the comments.

The part with "Nothing is broken, are you kidding me?" refers to the part where he got killed by guards after pvp-ing when he was expecting to get help from them. That is exactly how it should happen, no bugs there.

Patch 3.0.1:
"This seems to happen especially bad on my server, the other way around though, alliance sit just outside the aggro range of the guards and dot, cast, ranged attack me even though I can stand directly on the guards who do absolutely nothing. " - this looks as if this was an exception from the normally expected behavior of the guards. Again, this is only for the Quel-Dalas guards.

"The way it was supposed to work:
The plan was to get the guards to kill all those who are attacking each other inside or nearby the village.

How it works:
The guards SOMETIMES attack those who attack you in the village, but always attack you, no matter what you've done (at least that how it is to me :P)

What to do:
Keep away from the guards as long as you are in a battle. They consider you hostile if you run to them while in a PvP fight, while they don't as long as you're attacked while being with them." - this is.... come on, read it. It confirms here basically what i'm saying about the target/source players. This was the exception, all other guards simply protected you if you were in their range.

**Comment by Phoosy** [comment]
>*Obviously not up to snuff in terms of actual evidence*

You seemed to miss that part. I wouldn't think the guards in one area would operate differently in another area, but what do I know?

Also, you quoting a post that states the way it is supposed to work without citing where they pulled that data from is about as useful as my own post, unfortunately.

**Comment by Cabro** [comment]
The quotes are from the same thread of comments you linked before my comment. Just a few comments above. Scroll up a bit.

**Comment by Phoosy** [comment]
I'm aware, hence why I didn't bother including it because it was posturing what things should/should not be rather than someone relaying actual experience with the mechanic. I don't care what they _think_ it should be, only what it was/is, especially they're making a claim that it should be a certain way without citing _something_ of note. Kind of how we're supposed to do that.

**Comment by Cabro** [comment]
Ye, when I will have time this weekend I'm gonna watch hella videos and give link proofs to all my bug reports. I understand the need for you guys requiring proof and i respect it. Just leave this bug report open :)
